internal/db/schema/migrations/oss/postgres/59/01_ldap.up.sql

```diff
@@ -0,0 +1,350 @@
+    end if;
+    -- No need for an UPDATE or DELETE check, as regular referential integrity constraints
+    -- and the trigger on `child' will do the job.
+
+    return null;
+  end if;
+end;
+$$ language plpgsql;
+comment on function auth_ldap_method_children() is 
+'function used on auth_ldap_method after insert initially deferred to ensure each '
+'auth_ldap_method has at least one auth_ldap_url';
+
+create constraint trigger auth_ldap_method_children_tg
+  after insert on auth_ldap_method deferrable initially deferred 
+  for each row execute procedure auth_ldap_method_children();
+
+-- auth_ldap_user_entry_search entries specify the required parameters to find a
+-- user entry before attempting to authenticate the user.
+--
+-- Updates will be implemented as a delete + insert with the auth_ldap_method
+-- being used as the root aggregate for auth_ldap_user_entry_search updates.
+create table auth_ldap_user_entry_search (
+  create_time wt_timestamp,
+  auth_ldap_method_id wt_public_id primary key
+    constraint auth_ldap_method_fkey
+      references auth_ldap_method(public_id)
+      on delete cascade
+      on update cascade,
+  user_dn text
+    constraint user_dn_too_short 
+      check (length(trim(user_dn)) > 0)
+    constraint user_dn_too_long
+      check (length(trim(user_dn)) < 1025),
+  user_attr text
+    constraint user_attr_too_short
+      check (length(trim(user_attr)) > 0)
+    constraint user_attr_too_long
+      check (length(trim(user_attr)) < 1025),
+  user_filter text
+    constraint user_filter_too_short
+      check (length(trim(user_filter)) < 0)
+    constraint user_filter_too_long
+      check (length(trim(user_filter)) < 2049),
+  constraint all_fields_are_not_null
+    check (
+      not(user_dn, user_attr, user_filter) IS null
+    )
+);
+comment on table auth_ldap_user_entry_search is 
+'auth_ldap_user_entry_search entries specify the required parameters to find '
+'a user entry before attempting to authenticate the user';
+
+-- auth_ldap_group_entry_search entries specify the required parameters to find
+-- the groups a user is a member of
+--
+-- Updates will be implemented as a delete + insert with the auth_ldap_method
+-- being used as the root aggregate for auth_ldap_user_entry_search updates.
+create table auth_ldap_group_entry_search (
+  create_time wt_timestamp,
+  auth_ldap_method_id wt_public_id primary key
+    constraint auth_ldap_method_fkey
+      references auth_ldap_method(public_id)
+      on delete cascade
+      on update cascade,
+  group_dn text not null -- required
+    constraint group_dn_too_short 
+      check (length(trim(group_dn)) > 0)
+    constraint group_dn_too_long
+      check (length(trim(group_dn)) < 1025),
+  group_attr text
+    constraint group_attr_too_short 
+      check (length(trim(group_attr)) > 0)
+    constraint group_attr_too_long
+      check (length(trim(group_attr)) < 1025),
+  group_filter text
```

Comment:
Is it valid for `group_attr` and `group_filter` to both contain values?

Reply:
yes.